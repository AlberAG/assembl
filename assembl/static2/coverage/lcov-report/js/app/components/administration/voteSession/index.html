<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/administration/voteSession</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> js/app/components/administration/voteSession
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">24.52% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>76/310</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.93% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>38/112</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">14.84% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>23/155</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>73/282</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="customizeGaugeForm.jsx"><a href="customizeGaugeForm.jsx.html">customizeGaugeForm.jsx</a></td>
	<td data-value="12.5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="56" class="abs low">7/56</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="8.57" class="pct low">8.57%</td>
	<td data-value="35" class="abs low">3/35</td>
	<td data-value="15.56" class="pct low">15.56%</td>
	<td data-value="45" class="abs low">7/45</td>
	</tr>

<tr>
	<td class="file low" data-value="gaugeForm.jsx"><a href="gaugeForm.jsx.html">gaugeForm.jsx</a></td>
	<td data-value="14.55" class="pic low"><div class="chart"><div class="cover-fill" style="width: 14%;"></div><div class="cover-empty" style="width:86%;"></div></div></td>
	<td data-value="14.55" class="pct low">14.55%</td>
	<td data-value="55" class="abs low">8/55</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="16" class="abs medium">8/16</td>
	<td data-value="9.52" class="pct low">9.52%</td>
	<td data-value="21" class="abs low">2/21</td>
	<td data-value="14.81" class="pct low">14.81%</td>
	<td data-value="54" class="abs low">8/54</td>
	</tr>

<tr>
	<td class="file low" data-value="modulesSection.jsx"><a href="modulesSection.jsx.html">modulesSection.jsx</a></td>
	<td data-value="24.53" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.53" class="pct low">24.53%</td>
	<td data-value="53" class="abs low">13/53</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="18" class="abs low">8/18</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="27" class="abs low">6/27</td>
	<td data-value="21.28" class="pct low">21.28%</td>
	<td data-value="47" class="abs low">10/47</td>
	</tr>

<tr>
	<td class="file medium" data-value="numberGaugeForm.jsx"><a href="numberGaugeForm.jsx.html">numberGaugeForm.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="pageForm.jsx"><a href="pageForm.jsx.html">pageForm.jsx</a></td>
	<td data-value="54.55" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 54%;"></div><div class="cover-empty" style="width:46%;"></div></div></td>
	<td data-value="54.55" class="pct medium">54.55%</td>
	<td data-value="22" class="abs medium">12/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="textGaugeForm.jsx"><a href="textGaugeForm.jsx.html">textGaugeForm.jsx</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="tokenCategoryForm.jsx"><a href="tokenCategoryForm.jsx.html">tokenCategoryForm.jsx</a></td>
	<td data-value="28.57" class="pic low"><div class="chart"><div class="cover-fill" style="width: 28%;"></div><div class="cover-empty" style="width:72%;"></div></div></td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	</tr>

<tr>
	<td class="file low" data-value="tokensForm.jsx"><a href="tokensForm.jsx.html">tokensForm.jsx</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="24" class="abs low">6/24</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="10" class="abs low">3/10</td>
	<td data-value="26.09" class="pct low">26.09%</td>
	<td data-value="23" class="abs low">6/23</td>
	</tr>

<tr>
	<td class="file low" data-value="voteProposalForm.jsx"><a href="voteProposalForm.jsx.html">voteProposalForm.jsx</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="64" class="abs low">16/64</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="48" class="abs low">12/48</td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="32" class="abs low">2/32</td>
	<td data-value="27.59" class="pct low">27.59%</td>
	<td data-value="58" class="abs low">16/58</td>
	</tr>

<tr>
	<td class="file low" data-value="voteProposalsSection.jsx"><a href="voteProposalsSection.jsx.html">voteProposalsSection.jsx</a></td>
	<td data-value="35.71" class="pic low"><div class="chart"><div class="cover-fill" style="width: 35%;"></div><div class="cover-empty" style="width:65%;"></div></div></td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="14" class="abs low">5/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="38.46" class="pct low">38.46%</td>
	<td data-value="13" class="abs low">5/13</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
