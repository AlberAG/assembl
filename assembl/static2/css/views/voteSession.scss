.votesession-page {
  .header-content {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
  }

  .vote-instructions {
    padding: 0 20px;
    font-size: $font-size-m;
    font-family: $font-family4;
    max-width: 780px;
    text-align: justify;
    line-height: 32px;
  }

  .content-section {
    padding: 0;
  }

  .available-tokens {
    margin: 20px 0 40px;

    .category-available-tokens {
      display: flex;
      padding: 30px 20px;
      max-width: 780px;

      .tokens {
        padding-left: 50px;
        padding-top: 22px;
      }
    }
  }

  .available-tokens-sticky {
    position: fixed;
    top: 60px;
    width: 100%;
    z-index: 100;
    display: flex;
    justify-content: space-around;
    box-shadow: 3px 4px 7px #bbb;
    padding: 0 20px 12px;
    border-top: 1px solid $grey1;

    .separator {
      display: none;
    }

    .category-available-tokens {
      display: flex;

      .tokens {
        padding-left: 30px;
        padding-top: 20px;
      }
    }
  }

  .extra-margin-top {
    margin-top: 400px;
  }

  .tokens > * {
    margin: 5px;
    width: 30px;
    height: 30px;
  }

  .theme-box {
    &:first-child {
      margin: -70px 0 0;
    }
  }

  .proposal.row {
    margin: 90px 10px 30px;

    .proposal-title {
      font-family: $font-family;
      margin-top: 0;
    }

    .text {
      text-align: justify;
      padding-right: 20px;
    }

    .proposal-vote-modules {
      padding-top: 25px;
    }

    .tokens-line {
      margin-top: 14px;
      margin-bottom: 30px;

      .tokens {
        margin-left: -5px;
      }
    }

    .tokens > button {
      border: none;
      padding: 0;
      background: none;

      &:focus {
        outline: none;
      }

      &:active {
        box-shadow: none;
      }
    }

    .tokens > .btn-default:hover {
      background-color: transparent;
    }
  }

  .gauge-vote-for-proposal {
    margin-top: 40px;

    p {
      margin-bottom: 40px;
      margin-left: -5px;
    }
  }

  .form-actions {
    margin-top: 50px;
    margin-bottom: 100px;
  }

  .link-button {
    border: none;
    font-size: $font-size-xs;
    color: $link-color;
    text-decoration: none;
    background-color: $grey0;
    padding: 15px;

    &:hover {
      color: $link-hover-color;
      text-decoration: none;
    }

    &:focus {
      outline: none;
    }

    &:active {
      box-shadow: none;
    }

    [class^='assembl-icon-']:before,
    [class*=' assembl-icon-']:before {
      font-size: $font-size-xxs;
      margin-left: 5px;
    }
  }

  .link-button:focus,
  .link-button:active,
  .link-button:hover {
    background-color: transparent;
  }

  .expand-row {
    margin: 0 40px;
  }

  .votes-in-progress {
    margin: 0 40px;
    padding: 0 15px 30px;

    .box {
      margin-top: 30px;
      height: 160px;
    }

    .participants-count {
      .assembl-icon-profil:before {
        font-size: 40px;
        margin-bottom: 12px;
      }
    }

    .doughnut-container {
      margin: 0 auto;
      height: 80px;
      width: 80px;
      padding: 10px 0 0;

      .doughnut {
        .circle {
          stroke-width: 22px;
        }
      }
    }
  }

  .expand-result {
    padding-bottom: 40px;
  }

  .gauge-box {
    padding: 30px 0 0;
  }

  .separator {
    border-bottom: 1px $btn-dark-border-style $grey1;
    max-width: 745px;
    margin-left: 18px;
  }
}

.token-tooltip {
  border-radius: 4px;

  .tooltip-inner {
    background: none;
  }
}

@media screen and (min-width: 768px) {
  .votesession-page {
    .proposal.row {
      padding: 31px 15px 15px 15px;
    }

    .participants-count {
      padding: 40px 0 0;
    }

    .tokens-box {
      padding: 30px 0 0;
    }
  }
}

@media screen and (max-width: 767px) {
  .votesession-page {
    .available-tokens-sticky {
      display: none;
    }

    .container {
      padding: 0;
      margin: 0;
    }

    .proposal.row {
      padding: 15px 0 0;
    }

    .participants-count {
      padding: 32px 0 0;
    }

    .tokens-box {
      padding: 15px 0 0;
    }
  }
}
