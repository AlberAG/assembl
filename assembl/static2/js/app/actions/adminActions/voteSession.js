// @flow
import * as actionTypes from '../actionTypes';

export const updateVoteSessionPageTitle = (locale: string, value: string): actionTypes.UpdateVoteSessionPageTitle => ({
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE_TITLE
});

export const updateVoteSessionPageSubtitle = (locale: string, value: string): actionTypes.UpdateVoteSessionPageSubtitle => ({
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE_SUBTITLE
});

export const updateVoteSessionPageInstructionsTitle = (
  locale: string,
  value: string
): actionTypes.UpdateVoteSessionPageInstructionsTitle => ({
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE_INSTRUCTIONS_TITLE
});

export const updateVoteSessionPageInstructionsContent = (
  locale: string,
  value: string
): actionTypes.UpdateVoteSessionPageInstructionsContent => ({
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE_INSTRUCTIONS_CONTENT
});

export const updateVoteSessionPagePropositionsTitle = (
  locale: string,
  value: string
): actionTypes.UpdateVoteSessionPagePropositionsTitle => ({
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE_PROPOSITIONS_TITLE
});

export const updateVoteSessionHeaderImage = (value: File): actionTypes.UpdateVoteSessionHeaderImage => ({
  value: value,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE_IMAGE
});

export const updateVoteModules = (voteModules: actionTypes.VoteModulesArray): actionTypes.UpdateVoteModules => ({
  voteModules: voteModules,
  type: actionTypes.UPDATE_VOTE_MODULES
});

export const createTokenVoteModule = (id: string): actionTypes.CreateTokenVoteModule => ({
  id: id,
  type: actionTypes.CREATE_TOKEN_VOTE_MODULE
});

export const deleteTokenVoteModule = (id: string): actionTypes.DeleteTokenVoteModule => ({
  id: id,
  type: actionTypes.DELETE_TOKEN_VOTE_MODULE
});

export const updateTokenVoteExclusiveCategory = (id: string, value: boolean): actionTypes.UpdateTokenVoteExclusiveCategory => ({
  id: id,
  value: value,
  type: actionTypes.UPDATE_TOKEN_VOTE_EXCLUSIVE_CATEGORY
});

export const updateTokenVoteInstructions = (
  id: string,
  locale: string,
  value: string
): actionTypes.UpdateTokenVoteInstructions => ({
  id: id,
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_TOKEN_VOTE_INSTRUCTIONS
});

export const createTokenVoteCategory = (id: string, parentId: string): actionTypes.CreateTokenVoteCategory => ({
  id: id,
  parentId: parentId,
  type: actionTypes.CREATE_TOKEN_VOTE_CATEGORY
});

export const deleteTokenVoteCategory = (value: number, parentId: string): actionTypes.DeleteTokenVoteCategory => ({
  parentId: parentId,
  value: value,
  type: actionTypes.DELETE_TOKEN_VOTE_CATEGORY
});

export const updateTokenVoteCategoryTitle = (
  id: string,
  locale: string,
  value: string
): actionTypes.UpdateTokenVoteCategoryTitle => ({
  id: id,
  locale: locale,
  value: value,
  type: actionTypes.UPDATE_TOKEN_VOTE_CATEGORY_TITLE
});

export const updateTokenTotalNumber = (id: string, value: number): actionTypes.UpdateTokenTotalNumber => ({
  id: id,
  value: value,
  type: actionTypes.UPDATE_TOKEN_TOTAL_NUMBER
});

export const updateTokenVoteCategoryColor = (id: string, value: string): actionTypes.UpdateTokenVoteCategoryColor => ({
  id: id,
  value: value,
  type: actionTypes.UPDATE_TOKEN_VOTE_CATEGORY_COLOR
});

export const updateVoteSessionPage = (value: Object): actionTypes.UpdateVoteSessionPage => ({
  titleEntries: value.titleEntries,
  subTitleEntries: value.subTitleEntries,
  instructionsSectionTitleEntries: value.instructionsSectionTitleEntries,
  instructionsSectionContentEntries: value.instructionsSectionContentEntries,
  propositionsSectionTitleEntries: value.propositionsSectionTitleEntries,
  headerImage: value.headerImage,
  type: actionTypes.UPDATE_VOTE_SESSION_PAGE
});