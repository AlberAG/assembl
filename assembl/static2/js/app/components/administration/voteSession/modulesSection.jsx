// @flow
import React from 'react';
import { I18n, Translate } from 'react-redux-i18n';
import { connect } from 'react-redux';
import { Checkbox, SplitButton, MenuItem, Radio } from 'react-bootstrap';
import range from 'lodash/range';
import SectionTitle from '../sectionTitle';
import Helper from '../../common/helper';
import TokensForm from './tokensForm';
import GaugeForm from './gaugeForm';
import {
  createTokenVoteModule,
  createGaugeVoteModule,
  deleteVoteModule,
  updateVoteSessionPageSeeCurrentVotes
} from '../../../actions/adminActions/voteSession';

type ModulesSectionProps = {
  tokenModules: Object,
  gaugeModules: Object,
  editLocale: string,
  handleTokenCheckBoxChange: Function,
  handleGaugeCheckBoxChange: Function,
  handleGaugeSelectChange: Function,
  handleSeeCurrentVotesChange: Function,
  seeCurrentVotes: boolean
};

const DumbModulesSection = ({
  tokenModules,
  editLocale,
  handleTokenCheckBoxChange,
  gaugeModules,
  handleGaugeCheckBoxChange,
  handleGaugeSelectChange,
  handleSeeCurrentVotesChange,
  seeCurrentVotes
}: ModulesSectionProps) => {
  const tokenModuleChecked = tokenModules.size > 0;
  const gaugeModuleChecked = gaugeModules.size > 0;
  const tModule = tokenModules.toJS();
  const gModule = gaugeModules.toJS();
  const newId = Math.round(Math.random() * -1000000).toString();

  return (
    <div className="admin-box">
      <SectionTitle title={I18n.t('administration.voteSession.1')} annotation={I18n.t('administration.annotation')} />
      <div className="admin-content">
        <div className="form-container">
          <div className="vote-modules-form">
            <Checkbox
              checked={tokenModuleChecked}
              onChange={() => {
                handleTokenCheckBoxChange(tokenModuleChecked, tModule[0], newId);
              }}
            >
              <Helper
                label={I18n.t('administration.voteWithTokens')}
                helperUrl="/static2/img/helpers/helper4.png"
                helperText={I18n.t('administration.tokenVoteCheckbox')}
                classname="inline checkbox-title"
              />
            </Checkbox>
            {tokenModules.map(id => <TokensForm key={id} id={id} editLocale={editLocale} />)}
            <Checkbox
              checked={gaugeModuleChecked}
              onChange={() => {
                handleGaugeCheckBoxChange(gaugeModuleChecked, gModule, newId);
              }}
            >
              <Helper
                label={I18n.t('administration.voteWithGauges')}
                helperUrl="/static2/img/helpers/helper3.png" // TODO: add an actual screenshot
                helperText={I18n.t('administration.gaugeVoteCheckbox')}
                classname="inline checkbox-title"
              />
            </Checkbox>
            {gaugeModuleChecked ? (
              <div>
                <div className="flex">
                  <label htmlFor="input-dropdown-addon">
                    <Translate value="administration.gaugeNumber" />
                  </label>
                  <Helper helperUrl="/static2/img/helpers/helper2.png" helperText={I18n.t('administration.defineGaugeNumer')} />
                </div>
                <SplitButton
                  title={gaugeModules.size}
                  id="input-dropdown-addon"
                  required
                  onSelect={(eventKey) => {
                    handleGaugeSelectChange(eventKey, gaugeModules.size, newId, gModule);
                  }}
                >
                  {range(10).map(value => (
                    <MenuItem key={`gauge-item-${value + 1}`} eventKey={value + 1}>
                      {value + 1}
                    </MenuItem>
                  ))}
                </SplitButton>
              </div>
            ) : null}
            {gaugeModules.map((id, index) => <GaugeForm key={id} index={index} id={id} editLocale={editLocale} />)}
            <div className="margin-m">
              <label htmlFor="seeCurrentVotes">
                <Translate value="administration.seeCurrentVotes" />
              </label>
              <Radio
                id="seeCurrentVotes"
                onChange={() => {
                  handleSeeCurrentVotesChange(true);
                }}
                checked={seeCurrentVotes}
                name="seeCurrentVotes"
              >
                <Translate value="yes" />
              </Radio>
              <Radio
                onChange={() => {
                  handleSeeCurrentVotesChange(false);
                }}
                checked={!seeCurrentVotes}
                name="seeCurrentVotes"
              >
                <Translate value="no" />
              </Radio>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = ({ admin }) => {
  const { modulesInOrder, modulesById, page } = admin.voteSession;
  const { editLocale } = admin;
  return {
    tokenModules: modulesInOrder.filter(
      id => modulesById.getIn([id, 'type']) === 'tokens' && !modulesById.getIn([id, '_toDelete'])
    ),
    gaugeModules: modulesInOrder.filter(
      id => modulesById.getIn([id, 'type']) === 'gauge' && !modulesById.getIn([id, '_toDelete'])
    ),
    editLocale: editLocale,
    seeCurrentVotes: page.get('seeCurrentVotes')
  };
};

const mapDispatchToProps = dispatch => ({
  handleTokenCheckBoxChange: (checked, id, newId) => {
    if (!checked) {
      dispatch(createTokenVoteModule(newId));
    } else {
      dispatch(deleteVoteModule(id));
    }
  },
  handleGaugeCheckBoxChange: (checked, idArray, newId) => {
    if (!checked) {
      dispatch(createGaugeVoteModule(newId));
    } else {
      idArray.forEach((id) => {
        dispatch(deleteVoteModule(id));
      });
    }
  },
  handleGaugeSelectChange: (selectedNumber, gaugeNumber, newId, idArray) => {
    if (selectedNumber > gaugeNumber) {
      const numberToCreate = selectedNumber - gaugeNumber;
      for (let i = 0; i < numberToCreate; i += 1) {
        dispatch(createGaugeVoteModule(newId + i));
      }
    } else {
      idArray.forEach((id, index) => {
        const numberToDelete = gaugeNumber - selectedNumber;
        if (numberToDelete > index) {
          dispatch(deleteVoteModule(id));
        }
      });
    }
  },
  handleSeeCurrentVotesChange: checked => dispatch(updateVoteSessionPageSeeCurrentVotes(checked))
});

export { DumbModulesSection };

export default connect(mapStateToProps, mapDispatchToProps)(DumbModulesSection);