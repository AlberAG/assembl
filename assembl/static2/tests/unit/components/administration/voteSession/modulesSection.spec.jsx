import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { List } from 'immutable';

import { DumbModulesSection } from '../../../../../js/app/components/administration/voteSession/modulesSection';

describe('ModulesSection component', () => {
  const handleCheckBoxChangeSpy = jest.fn(() => {});

  it('should render a ModulesSection component without any TokensForm nor GaugeForm', () => {
    const props = {
      tokenModules: List(),
      gaugeModules: List(),
      editLocale: 'fr',
      handleCheckBoxChange: handleCheckBoxChangeSpy
    };
    const shallowRender = new ShallowRenderer();
    shallowRender.render(<DumbModulesSection {...props} />);
    const result = shallowRender.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  // TODO: fake a list of token modules to test their rendering
});