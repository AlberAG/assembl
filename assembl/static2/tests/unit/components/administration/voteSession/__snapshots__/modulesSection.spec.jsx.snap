// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModulesSection component should render a ModuleSection component with token and gauges 1`] = `
<div
  className="admin-box"
>
  <SectionTitle
    annotation="Missing translation: administration.annotation"
    title="Missing translation: administration.voteSession.1"
  />
  <div
    className="admin-content"
  >
    <div
      className="form-container"
    >
      <div
        className="vote-modules-form"
      >
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          <Helper
            additionalTextClasses=""
            classname="inline checkbox-title"
            helperText="Missing translation: administration.tokenVoteCheckbox"
            helperUrl="/static2/img/helpers/helper4.png"
            label="Missing translation: administration.voteWithTokens"
          />
        </Checkbox>
        Immutable.List [
        ]
        <Checkbox
          bsClass="checkbox"
          checked={true}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          <Helper
            additionalTextClasses=""
            classname="inline checkbox-title"
            helperText="Missing translation: administration.gaugeVoteCheckbox"
            helperUrl="/static2/img/helpers/helper3.png"
            label="Missing translation: administration.voteWithGauges"
          />
        </Checkbox>
        <div>
          <div
            className="flex"
          >
            <label
              htmlFor="input-dropdown-addon"
            >
              <Translate
                value="administration.gaugeNumber"
              />
            </label>
            <Helper
              additionalTextClasses=""
              classname=""
              helperText="Missing translation: administration.defineGaugeNumer"
              helperUrl="/static2/img/helpers/helper2.png"
              label=""
            />
          </div>
          <SplitButton
            id="input-dropdown-addon"
            onSelect={[Function]}
            required={true}
            title={1}
          >
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={1}
              header={false}
            >
              1
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={2}
              header={false}
            >
              2
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={3}
              header={false}
            >
              3
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={4}
              header={false}
            >
              4
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={5}
              header={false}
            >
              5
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={6}
              header={false}
            >
              6
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={7}
              header={false}
            >
              7
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={8}
              header={false}
            >
              8
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={9}
              header={false}
            >
              9
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey={10}
              header={false}
            >
              10
            </MenuItem>
          </SplitButton>
        </div>
        Immutable.List [
          <Connect(DumbGaugeForm)
            editLocale="en"
            id="my-num-gauge"
            index={0}
          />,
        ]
        <div
          className="margin-m"
        >
          <label
            htmlFor="seeCurrentVotes"
          >
            <Translate
              value="administration.seeCurrentVotes"
            />
          </label>
          <Radio
            bsClass="radio"
            checked={true}
            disabled={false}
            id="seeCurrentVotes"
            inline={false}
            name="seeCurrentVotes"
            onChange={[Function]}
            title=""
          >
            <Translate
              value="yes"
            />
          </Radio>
          <Radio
            bsClass="radio"
            checked={false}
            disabled={false}
            inline={false}
            name="seeCurrentVotes"
            onChange={[Function]}
            title=""
          >
            <Translate
              value="no"
            />
          </Radio>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModulesSection component should render a ModulesSection component without any TokensForm nor GaugeForm 1`] = `
<div
  className="admin-box"
>
  <SectionTitle
    annotation="Missing translation: administration.annotation"
    title="Missing translation: administration.voteSession.1"
  />
  <div
    className="admin-content"
  >
    <div
      className="form-container"
    >
      <div
        className="vote-modules-form"
      >
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          <Helper
            additionalTextClasses=""
            classname="inline checkbox-title"
            helperText="Missing translation: administration.tokenVoteCheckbox"
            helperUrl="/static2/img/helpers/helper4.png"
            label="Missing translation: administration.voteWithTokens"
          />
        </Checkbox>
        Immutable.List [
        ]
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          <Helper
            additionalTextClasses=""
            classname="inline checkbox-title"
            helperText="Missing translation: administration.gaugeVoteCheckbox"
            helperUrl="/static2/img/helpers/helper3.png"
            label="Missing translation: administration.voteWithGauges"
          />
        </Checkbox>
        Immutable.List [
        ]
        <div
          className="margin-m"
        >
          <label
            htmlFor="seeCurrentVotes"
          >
            <Translate
              value="administration.seeCurrentVotes"
            />
          </label>
          <Radio
            bsClass="radio"
            checked={false}
            disabled={false}
            id="seeCurrentVotes"
            inline={false}
            name="seeCurrentVotes"
            onChange={[Function]}
            title=""
          >
            <Translate
              value="yes"
            />
          </Radio>
          <Radio
            bsClass="radio"
            checked={true}
            disabled={false}
            inline={false}
            name="seeCurrentVotes"
            onChange={[Function]}
            title=""
          >
            <Translate
              value="no"
            />
          </Radio>
        </div>
      </div>
    </div>
  </div>
</div>
`;
